lib/livebook/notebook_manager.ex

```diff
@@ -207,6 +205,20 @@ defmodule Livebook.NotebookManager do
     {:noreply, state, {:continue, :dump_state}}
   end
 
+  @impl true
+  def handle_info({event, _file_system}, _state)
+      when event in [:file_system_created, :file_system_updated, :file_system_deleted] do
+    {:noreply, load_state()}
+  end
+
+  def handle_info({:hub_connected, _id}, _state) do
```

Comment:
This would not work in certain cases, for example if we have two hubs, then we wait for either to connect, but we still don't have file systems from the other one. But even with a single hub, if we are offline and can't connect, it shouldn't disable `NotebookManager`, especially that there are file from personal hub.

I think what we need is to store the hub file systems in storage, restore from there when starting teams client, and then once connected we diff as usual. We may have discussed this in the past, but I don't remember the details :)

Reply:
The way we run the migration, it would block the NotebookManager to start, because we're trying to get the file system list before all hubs are connected, which was the reason of this change.

In that case, we would need to start the NotebookManager after we call `connect_hubs/0` and run the migrations.

Reply:
The migration is only relevant for personal hub, because that's the only one people currently have, and those file systems we have on boot :)

Reply:
As we discussed internally, I'll rollback all the changes I did to NotebookManager so we can work on this issue separately.

Reply:
As per our call, we will revert the changes to the notebook manager. :)

Reply:
The only thing I'll keep changed is to get the file systems only from Personal hub, so the behaviour will keep as-is